L1Trigger/TrackFindingTracklet/src/PurgeDuplicate.cc

```diff
@@ -843,3 +831,24 @@ bool PurgeDuplicate::isTrackInBin(const std::vector<unsigned int>& vec, unsigned
   bool found = (result != vec.end());
   return found;
 }
+
+void PurgeDuplicate::doCompareBest(const std::vector<std::pair<int, int>>& stubsTrk,
+                                   const std::vector<const Stub*>& fullStubslistsTrk,
+                                   int layStubidsTrk[],
+                                   unsigned int itrk) const {
+  for (unsigned int stcount = 0; stcount < stubsTrk.size(); stcount++) {
+    int i = stubsTrk[stcount].first;  // layer/disk
+    bool barrel = (i > 0 && i < 10);
```

Comment:
Magic numbers like 10 are not allowed in CMSSW

Reply:
I feel like we've had this discussion on these lines before. From what I can see, the 10 is used because the disk layers are +/- 11-15. I can change 10 in the first line to`barrel = ( i > 0 && i <= N_LAYER) ` and then in the second line to `endcapA = i > N_LAYER' and `endcapB = i < 0`. However, the (i - 1) and (i - 5) terms in the line that sets `lay` are there specifically to encode the layers to 0-15.